src/Symfony/Component/Validator/Command/DebugCommand.php

```diff
@@ -0,0 +1,192 @@
+
+            return 0;
+        }
+
+        foreach ($this->getResourcesByPath($class) as $class) {
+            $this->dumpValidatorsForClass($input, $output, $class);
+        }
+
+        return 0;
+    }
+
+    private function dumpValidatorsForClass(InputInterface $input, OutputInterface $output, string $class): void
+    {
+        $io = new SymfonyStyle($input, $output);
+        $title = sprintf('Validation information for class "<info>%s</info>"', $class);
+        $rows = [];
+        $dump = new Dumper($output);
+
+        foreach ($this->getConstrainedPropertiesData($class) as $propertyName => $constraintsData) {
+            foreach ($constraintsData as $data) {
+                $rows[] = [
+                    $propertyName,
+                    $data['class'],
+                    implode(', ', $data['groups']),
+                    $dump($data['options']),
+                ];
+            }
+        }
+
+        if (empty($rows)) {
+            if (false === $input->getOption('show-all')) {
+                return;
+            }
+
+            $io->title($title);
+            $io->text('No validators were found for this class.');
+
+            return;
+        }
+
+        $io->title($title);
+
+        $table = new Table($output);
+        $table->setHeaders(['Property', 'Name', 'Groups', 'Options']);
+        $table->setRows($rows);
+        $table->setColumnMaxWidth(3, 80);
+        $table->render();
+    }
+
+    private function getConstrainedPropertiesData(string $class): array
+    {
+        $data = [];
+
+        /** @var ClassMetadataInterface $classMetadata */
+        $classMetadata = $this->validator->getMetadataFor($class);
+
+        foreach ($classMetadata->getConstrainedProperties() as $constrainedProperty) {
+            $data[$constrainedProperty] = $this->getPropertyData($classMetadata, $constrainedProperty);
+        }
+
+        return $data;
+    }
+
+    private function getPropertyData(ClassMetadataInterface $classMetadata, string $constrainedProperty): array
+    {
+        $data = [];
+
+        $propertyMetadata = $classMetadata->getPropertyMetadata($constrainedProperty);
+        foreach ($propertyMetadata as $metadata) {
+            foreach ($metadata->getConstraints() as $constraint) {
+                $data[] = [
+                    'class' => \get_class($constraint),
+                    'groups' => $constraint->groups,
+                    'options' => $this->getConstraintOptions($constraint),
+                ];
+            }
+        }
+
+        return $data;
+    }
+
+    private function getConstraintOptions(Constraint $constraint): array
+    {
+        $options = [];
+
+        $constraintClass = \get_class($constraint);
+
+        foreach (array_keys(get_class_vars($constraintClass)) as $propertyName) {
+            try {
+                $options[$propertyName] = $constraint->$propertyName;
+            } catch (\Exception $exception) {
```

Comment:
Why is this catch required?

Reply:
It can throw an Symfony\Component\Validator\Exception\InvalidOptionException for example with 
static property $validationModes on Symfony\Component\Validator\Constraints.
Maybe, we could find something better.

Reply:
Oh, got it. Indeed, we should not try to access the static properties. I meant [`get_object_vars`](https://www.php.net/manual/fr/function.get-object-vars.php) in my previous suggestion.
It's actually the function used by the Constraint class itself.

Reply:
It dumps "groups" which are dynamically added. But I skipped that groups cause we have these information in another column.